v5/pivotal/aggregator.go

```diff
@@ -0,0 +1,133 @@
+package pivotal
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+const aggregatorURL = "https://www.pivotaltracker.com/services/v5/aggregator"
+
+type AggregatorService struct {
+	client             *Client
+	requests           []string
+	aggregatedResponse map[string]interface{} // Storing the response temporarily.
+}
+
+func newAggregatorService(client *Client) *AggregatorService {
+	return &AggregatorService{client, []string{}, nil}
+}
+
+// For making sure the request is in empty state
+func (service *AggregatorService) InitRequest() {
+	service.client.Aggregator.requests = []string{}
+	service.client.Aggregator.aggregatedResponse = make(map[string]interface{})
+}
+
+func (service *AggregatorService) QueueStoryCommentsByID(projectID, storyID int) {
```

Comment:
This pattern seems different than anything else this package does. In general, I would recommend making APIs immutable whenever possible as it reduces the chance for errors. In particular, this design seems prone to race conditions if used from multiple goroutines and requires a somewhat awkward `InitRequest` and `FinishRequest`. I think this could be cleaner if you built up the request using the builder pattern or just made an `GetAggregate` (or whatever) method that had variadic arguments of a particular type (e.g., [functional options](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis)).

The value returned by this method could then have the getter methods you have below.

I think in this case, that would mean something

Reply:
Yes, I didn't think about that. I have changed it to a builder model. What do you think?

Now below is how it would be used:
```
ptClient := pivotal.NewClient("TOKEN")

storiesToGet := []int{
    181915841,
    182087301,
    182337490,
    182276377,
    181814584,
}
projectID := 557367
aggregation, _ := ptClient.Aggregator.BuildStoriesCommentsAndReviewsFor(projectID, storiesToGet)

story, _ := aggregation.GetStory(181915841)
comments, _ := aggregation.GetComments(181915841)
reviews, _ := aggregation.GetReviews(181915841)
```

Reply:
I have changed it. I've also added getting the stories by project ID and story ID.
Here is how it works now:
```
	ptClient := pivotal.NewClient("TOKEN")

	storiesToGet := []int{
		181915841,
		182087301,
		182337490,
		182276377,
	}
	projectID := 557367
	aggregation := ptClient.Aggregator.GetBuilder().Stories(projectID, storiesToGet).Story(projectID, 181814584)
	aggregation.CommentsOfStories(projectID, storiesToGet).ReviewsOfStories(projectID, storiesToGet)
	aggregation, err := aggregation.Send()

	story, _ := aggregation.GetStory(projectID, 181915841)
	comments, _ := aggregation.GetComments(projectID, 181915841)
	reviews, _ := aggregation.GetReviews(projectID, 181915841)
```